packages/number-input/src/NumberInput.story.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useRef, useState } from 'react';
+import { Story } from '@storybook/react';
+
+import { css } from '@leafygreen-ui/emotion';
+import { storybookArgTypes, StoryMeta } from '@leafygreen-ui/lib';
+
+import {
+  NumberInputProps,
+  Size,
+  State,
+  UnitOption,
+} from './NumberInput/NumberInput.types';
+import { NumberInput } from '.';
+
+export default StoryMeta({
+  title: 'Components/NumberInput',
+  component: NumberInput,
+  args: {
+    label: 'label',
+    unitOptions: [],
+    disabled: false,
+  },
+  argTypes: {
+    darkMode: storybookArgTypes.darkMode,
+    size: {
+      control: 'select',
+      options: Object.values(Size),
+    },
+    state: {
+      control: 'select',
+      options: Object.values(State),
+    },
+    disabled: {
+      control: 'boolean',
+    },
+    label: {
+      control: 'text',
+    },
+    description: {
+      control: 'text',
+    },
+    unit: {
+      control: 'text',
+    },
+    unitOptions: {
+      control: 'none',
+    },
+    errorMessage: {
+      control: 'text',
+    },
+    placeholder: {
+      control: 'text',
+    },
+  },
+  parameters: {
+    default: 'Demo',
+    controls: {
+      exclude: [
+        'as',
+        'children',
+        'aria-labelledby',
+        'aria-describedby',
+        'aria-label',
+        'value',
+        'onSelectChange',
+      ],
+    },
+  },
+});
+
+const unitOptions = [
+  {
+    displayName: 'One(s)',
+    value: 'one',
+  },
+  {
+    displayName: 'Two(s)',
+    value: 'two',
+  },
+  {
+    displayName: 'Three(s)',
+    value: 'three',
+  },
+  {
+    displayName: 'FourFiveSixSeven(s)',
+    value: 'four',
+  },
+];
+
+type StoryProps = NumberInputProps & { view: string };
+
+const Template: Story<StoryProps> = args => {
+  const { view, unit: unitProp = 'one', ...rest } = args;
+  const [unit, setUnit] = useState<string>(unitOptions[0].displayName);
+  const [value, setValue] = useState<string>('');
+  const inputRef = useRef<HTMLInputElement | null>(null);
```

Comment:
Noticed that when the input is controlled, exponents (e.g. `2e4`) are not allowed. This doesn't quite match this sandbox (which also behaves somewhat oddly)
https://codesandbox.io/s/quiet-flower-8dcqwd?file=/src/App.tsx